### DISCUSSION TITLE
Audiophonics on/off implementation?

### DISCUSSION THREAD

**Discussion by PedroG64** [rern/rAudio#804] [Feature Requests]
Hello Rern,
rAudio is my best streamer audio after runeAudio. Your work if wonderful for me and the quality of your musical application.
thank you for sharing your work, I am unable to program.
Is it possible to implement the audiophonics on/off button which was integrable for the old Hondax RuneAudio? It is no longer functional for the latest developments.
I give you the summary of the Audiophonics site of the old method to do it, if ever possible.
It is for a DIY assembly with a clean stop system. 

> Tutoriel sur RuneAudio 0.4 :

1/ Installez RuneAudio 0.4 sur une carte SD / Use the latest image for RP2/RP3
Démarrez le pi, connectez vous avec putty.

2/ Mettre à jour RuneAudio / update RuneAudio

Code : [Tout sélectionner](http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81#)

cd /var/www
git pull

3/ Installer la librairie WiringPi qui permet de lire les état des pins GPIO / install wiringPi

Code : [Tout sélectionner](http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81#)

pacman -Sy wiringpi

Confirmer par "y"

4/ Télécharger le script sds.sh pour lire les états sur le GPIO / Download sds.sh script to read GPIO pins state

Code : [Tout sélectionner](http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81#)

wget -O /usr/local/bin/sds.sh "https://raw.githubusercontent.com/audiophonics/Raspberry-pwr-management/master/runeaudio.sds.sh"

5/ Donner les droits d'execution au fichier / make it executable

Code : [Tout sélectionner](http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81#)

chmod +x /usr/local/bin/sds.sh

6/ Ajouter l'extinction depuis la page Web de RuneAudio / change /var/www/command/rune_shutdown
On renomme le fichier rune_shutdown, et on télécharge le nouveau / Rename and download new rune_shutdown file

Code : [Tout sélectionner](http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81#)

mv /var/www/command/rune_shutdown /var/www/command/rune_shutdown.sav
wget -O /var/www/command/rune_shutdown "https://raw.githubusercontent.com/audiophonics/Raspberry-pwr-management/master/runeaudio.rune_shutdown"
chmod +x /var/www/command/rune_shutdown

Editer le fichier /var/www/command/rune_SY_wrk
Remplacer ligne 608 :

Code : [Tout sélectionner](http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81#)

                    sysCmd('/var/www/command/rune_shutdown');
                    sysCmdAsync('poweroff');

par

Code : [Tout sélectionner](http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81#)

                    sysCmd('/var/www/command/rune_shutdown poweroff');
                    sysCmdAsync('poweroff');

7/ Création d'un service pour executer sds.sh / make a service for sds.sh

Code : [Tout sélectionner](http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81#)

wget -O /usr/lib/systemd/system/sds.service "https://raw.githubusercontent.com/audiophonics/Raspberry-pwr-management/master/runeaudio.sds.service"

8/ Activation du service SDS / enable the service

Code : [Tout sélectionner](http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81#)

systemctl enable sds.service

Merci à Hondagx35 de RuneAudio pour l'adaptation 8)
[http://www.runeaudio.com/forum/audiopho ... tml#p13106](http://www.runeaudio.com/forum/audiophonics-i-sabre-v3-dac-es9023-tcxo-t3377.html#p13106)" onclick="window.open(this.href);return false;

> 
It's on this internet page:
http://forum.audiophonics.fr/viewtopic.php?f=4&t=1620&sid=619fd50c689d22e5f93b906a16961c81

Thanks in advance.

**Comment by rern** [comment]
Try:
```sh
echo "\
gpio -1 mode 7 in
gpio -1 mode 7 up
gpio -1 wfi 7 falling
gpio -1 mode 7 out
gpio -1 write 7 1
sleep 1
gpio -1 write 7 0
/srv/http/bash/cmd.sh power$'\n'off
" > button.sh
chmod +x button.sh
./button.sh &
```
Then press the button.

**Reply by PedroG64** [reply]
Hello Rern,
I tried your solution, it doesn't work.
Thanks for your effort
